src/pages/[platform]/build-a-backend/data/connect-event-api/index.mdx

```diff
@@ -27,19 +28,192 @@ export function getStaticProps(context) {
+
+*  `apiKey` authorization, **APIKeyAuthorizer**
+*  `identityPool` authorization, **IAMAuthorizer**
+*  `userPool` authorization, **AuthTokenAuthorizer**
+
+You can create as many Events clients as necessary if you require multiple authorization types.
+#### API KEY
+
+An `ApiKeyAuthorizer` can be used with a hardcoded API key or by fetching the key from some source.:
+
+```kotlin
+// highlight-start
+// Use a hard-coded API key
+val authorizer = ApiKeyAuthorizer("[API_KEY]")
+//highlight-end
+// or
+// highlight-start
+// Fetch the API key from some source. This function may be called many times,
+// so it should implement appropriate caching internally.
+val authorizer = ApiKeyAuthorizer { fetchApiKey() }
+//highlight-end
+```
+
+#### AMAZON COGNITO USER POOLS
+
+When working directly with AppSync, you must implement the token fetching yourself.
+
+```kotlin
+// highlight-start
+// Use your own token fetching. This function may be called many times,
+// so it should implement appropriate caching internally.
+val authorizer = AuthTokenAuthorizer {
+  fetchLatestAuthToken()
+}
+//highlight-end
+```
+
+#### AWS IAM
+
+When working directly with AppSync, you must implement the request signing yourself.
+
+```kotlin
+// highlight-start
+// Provide an implementation of the signing function. This function should implement the 
+// AWS Sig-v4 signing logic and return the authorization headers containing the token and signature.
+val authorizer = IamAuthorizer { appSyncRequest -> signRequestAndReturnHeaders(appSyncRequest) }
+// highlight-end
+```
+</Block>
+
+<Block name="With Amplify">
+
+The AWS AppSync Events library imports a number of Authorizer classes to match the various authorization strategies that may be used for your Events API. You should choose the appropriate Authorizer type for your authorization strategy.
+
+*  `apiKey` authorization, **APIKeyAuthorizer**
+*  `identityPool` authorization, **AmplifyIAMAuthorizer**
+*  `userPool` authorization, **AmplifyUserPoolAuthorizer**
+
+You can create as many Events clients as necessary if you require multiple authorization types.
+#### API KEY
+
+An `ApiKeyAuthorizer` can provide a hardcoded API key, or fetch the API key from some source:
+
+```kotlin
+// highlight-start
+// Use a hard-coded API key
+val authorizer = ApiKeyAuthorizer("[API_KEY]")
+//highlight-end
+// or
+// highlight-start
+// Fetch the API key from some source. This function may be called many times,
+// so it should implement appropriate caching internally.
+val authorizer = ApiKeyAuthorizer { fetchApiKey() }
+//highlight-end
+```
+
+#### AMAZON COGNITO USER POOLS
+
+```kotlin
+// highlight-start
+// Using the provided Amplify UserPool Authorizer
+val authorizer = AmplifyUserPoolAuthorizer()
+//highlight-end
+// or
+// highlight-start
+// Use your own token fetching. This function may be called many times,
+// so it should implement appropriate caching internally.
+val authorizer = AuthTokenAuthorizer {
```

Comment:
We can remove this example given this is the "with Amplify" section.

Reply:
Thats fair. There would be no reason to implement their own at that point.